examples/iterative_probabilistic_kh.py

```diff
@@ -0,0 +1,182 @@
+length `N`, in which case the iteration `i` is run with the probabilistic version of
+Kernel Herding with the temperature of `t_schedule[i]`.
+
+Generally, standard (deterministic) Kernel Herding converges very quickly so that
+`refine` keeps producing the same coreset after a few iterations (i.e., it finds a
+fixed point). Introducing probabilistic selection perturbs the procedure and helps
+find new coresets. The temperature parameter can help balance this trade-off: high
+values tend to produce random coresets, while low values approximate standard Kernel
+Herding a converge faster.
+"""
+
+from typing import Dict, Optional
+
+import jax
+import jax.numpy as jnp
+import matplotlib.pyplot as plt
+import numpy as np
+from jaxtyping import Array
+from sklearn.datasets import make_blobs
+
+from coreax import MMD, Coresubset, Data, SquaredExponentialKernel
+from coreax.kernels import median_heuristic
+from coreax.solvers import KernelHerding
+from coreax.solvers.coresubset import _initial_coresubset  # noqa: PLC2701
+
+
+def make_data(
+    num_data_points: int = 10_000,
+    num_features: int = 2,
+    num_cluster_centers: int = 10,
+    random_seed: int = 123,
+) -> Data:
+    """Create dataset using sklearn.datasets.make_blobs."""
+    x, *_ = make_blobs(
+        n_samples=num_data_points,
+        n_features=num_features,
+        centers=num_cluster_centers,
+        random_state=random_seed,
+        return_centers=True,
+        cluster_std=2,
+    )
+    return Data(jnp.asarray(x))
+
+
+def get_kernels(
+    data: Data, random_seed: int = 0
+) -> tuple[SquaredExponentialKernel, SquaredExponentialKernel]:
+    """Get a SquaredExponentialKernel based on the data using the median heuristic."""
+    num_data_points = len(data)
+    num_samples_length_scale = min(num_data_points, 1_000)
+    generator = np.random.default_rng(random_seed)
+    idx = generator.choice(num_data_points, num_samples_length_scale, replace=False)
+    length_scale = median_heuristic(data.data[idx])
+    return (
+        SquaredExponentialKernel(length_scale=length_scale),
+        SquaredExponentialKernel(
+            length_scale=length_scale,
+            output_scale=1.0 / (length_scale * jnp.sqrt(2.0 * jnp.pi)),
+        ),
+    )
+
+
+def iterative_refine_experiment(
+    data: Data,
+    coreset_size: int,
+    n_iter: int,
+    t_schedule: Optional[Array] = None,
+    seed: int = 0,
+) -> tuple[Array, Coresubset]:
+    """Perform an experiment with n_iter iterations and a given temperature schedule."""
+    random_key = jax.random.key(seed)
+    kernel, mmd_kernel = get_kernels(data)
+    mmd_metric = MMD(mmd_kernel)
+    probabilistic = True
+    if t_schedule is None:
+        t_schedule = jnp.ones(n_iter)
+        probabilistic = False
+
+    mmd_data = jnp.zeros(n_iter)  # store experiment data
+    initial_coreset = _initial_coresubset(0, coreset_size, data)
+
+    def run_exp(i, state):
```

Comment:
```suggestion
    def run_experiment(i, state):
```
No need to shorten the name here! Could also benefit from a short docstring or comment explaining what it does. Also, add type hints for the arguments and return type.